src/components/Chart/ChartOptions.jsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import React, { useEffect, useState } from 'react';
+import Dropdown from 'react-bootstrap/Dropdown';
+import Form from 'react-bootstrap/Form';
+import SelectableContext from 'react-bootstrap/SelectableContext';
+import { useDispatch, useSelector } from 'react-redux';
+import { NumberInlineInput } from 'pc-nrfconnect-shared';
+
+import { chartState, scaleChartYAxisAction } from '../../reducers/chartReducer';
+
+const unitLabels = ['\u00B5A', 'mA', 'A'];
+
+const ChartOptions = () => {
+    const { yMin, yMax, yAxisLock } = useSelector(chartState);
+    const dispatch = useDispatch();
+    const [localYMax, setLocalYMax] = useState(1);
+
+    /** Mapping to the power to get correct value of uA
+     * 0=uA, 1=mA, 2=A
+     */
+    const [unitPower, setUnitPower] = useState(0);
+
+    useEffect(() => {
+        const power = getUnitPower(yMax);
+        setUnitPower(power);
+        const precisionResult = Number(yMax / 1000 ** power).toPrecision(4);
+        const rawResult = Number(yMax / 1000 ** power);
+        setLocalYMax(
+            precisionResult.toString().length < rawResult.toString().length
```

Comment:
[minor] Maybe extract this to a function with a descriptive name since it's not immediately obvious what is going on here?

Reply:
Done :+1: